### PR TITLE
Response confirmation email

### PR DESCRIPTION
**Pull request by Carvalho96v2**
- Added emailing functionality to the response api's post method
- Added a common strings file for statically defined strings
- Mocked out the emailer service in debug environments

Please review if I mocked the emailer service out correctly

### CODE REVIEW

File: api/app/responses/api.py [hunk 1]
```diff
@@ -82,7 +83,41 @@ def post(self):
                 db.session.add(answer)
             db.session.commit()
 
-            return {'id': response.id, 'user_id': user_id}, 201  # 201 is 'CREATED' status code
+            try:
```
Comment (avishkar58) [inline api/app/responses/api.py]: Can we refactor this into another method (eg send_confirmation) and call it from the both POST and PUT, whenever response.is_submitted is TRUE (ie. we only send a confirmation mail when the user submits their response, not when they just save it for later).

File: api/app/responses/api.py [hunk 2]
```diff
@@ -82,7 +83,41 @@ def post(self):
...
+                user = db.session.query(AppUser).filter(AppUser.id==g.current_user['id']).first()
+                subject = strings.build_response_email_subject(user.user_title, user.firstname, user.lastname)
+
+                answers = db.session.query(Answer).filter(Answer.response_id == response.id).all()
+                if answers is None:
+                    LOGGER.warn('Found no answers associated with response with id {response_id}'.format(response_id=response.id))
```
Comment (avishkar58) [inline api/app/responses/api.py]: Thanks for adding logging to this controller! I forgot about that :)

File: api/app/utils/emailer.py [hunk 3]
```diff
@@ -22,43 +22,49 @@ def send_mail(recipient, subject, body_text, body_html='', charset='UTF-8', mail
...
-            LOGGER.info("Building Mimemultipart msg")
-            msg = MIMEMultipart()
-            LOGGER.info("Setting subject %s"%(subject,))
-            msg['Subject'] = subject
-            LOGGER.info("Setting Sender name : %s and Sender email %s"%(SMTP_SENDER_NAME, SMTP_SENDER_EMAIL))
-            msg['From'] = email.utils.formataddr((SMTP_SENDER_NAME, SMTP_SENDER_EMAIL))
-            LOGGER.info("Setting recipient %s"%(recipient))
-            msg['To'] = recipient
-            #msg.add_header('X-SES-CONFIGURATION-SET', configuration_set)
+    if (not DEBUG):
```
Comment (avishkar58) [inline api/app/utils/emailer.py]: Normally we'd do this by having an AmazonMailer and a MockMailer, both implementing the common send_mail interface and pass the appropriate one to the controllers (dependency injection), but that's a fair bit more complex, so this is fine for now!

File: api/app/responses/api.py (hunk 1 above)
Comment (Carvalho96v2) [reply api/app/responses/api.py]: Aha, gotcha :)

File: api/app/utils/emailer.py (hunk 3 above)
Comment (Carvalho96v2) [reply api/app/utils/emailer.py]: How would that piece of logic work? Where would you pass the mailer instance into the controller, and how would you know which one to pass?

**Comment by avishkar58** [comment]
Here's an article that describes how it can be done in Flask

http://christophergs.github.io/python/2018/09/25/elegant-flask-apis-pt-1/

But rather leave what you have for now - we can always refactor later if we
have more cases warranting the extra complexity

File: api/app/utils/emailer.py (hunk 3 above)
Comment (nischalhp) [reply api/app/utils/emailer.py]: Just adding to @avishkar58 thought process, the dependency injection can happen using various things, for example as simple as environment variables, where we can base the injection based on a value being set vs unset and inject the appropriate controller. We do not need it at this point in time, but its something that can be done later.

Comment (Carvalho96v2) [reply api/app/utils/emailer.py]: @avishkar58 @nischalhp Great, thanks guys! Will look definitely into it :)

**Pull request merged**